cmd/juju/commands/common.go

```diff
@@ -153,6 +154,34 @@ func resolveCharmStoreEntityURL(urlStr string, csParams charmrepo.NewCharmStoreP
 	return ref, repo, nil
 }
 
+func isTermAgreementRequired(err error) ([]string, bool) {
```

Comment:
Also, perhaps:
    // termsAgreementError returns err as a *termsAgreementError
    // if it has a "terms agreement request" error code, otherwise
    // it returns err.
    func termsAgreementError(err error) error

?